lib/bolt/config.rb

```diff
@@ -191,6 +195,7 @@ def initialize(project, config_data, overrides = {})
       ld_concurrency = loaded_data.map(&:keys).flatten.include?('concurrency')
       if default_concurrency != DEFAULT_DEFAULT_CONCURRENCY &&
          !ld_concurrency &&
+         !overrides[:from_default] &&
```

Comment:
I don't think that this is actually needed? The default config will never have its warnings emitted since it's replaced by the real config before that happens.

If we need this for some other reason I think it might be better to have it as a separate flag that's not part of the overrides hash, since that's technically supposed to be options passed from the CLI.